master/internal/api_auth.go

```diff
@@ -174,28 +174,27 @@ func processProxyAuthentication(c echo.Context) (done bool, err error) {
 	return err != nil, authz.SubIfUnauthorized(err, serviceNotFoundErr)
 }
 
-// processAuthWithRedirect is an auth middleware that redirects the requests
+// processAuthWithRedirect is an auth middleware that redirects browser requests
 // to login page for a set of given paths in case of authentication errors.
 func processAuthWithRedirect(redirectPaths []string) echo.MiddlewareFunc {
 	return func(next echo.HandlerFunc) echo.HandlerFunc {
 		return func(c echo.Context) error {
+			err := user.GetService().ProcessAuthentication(next)(c)
+			if err == nil {
+				return nil
+			}
+			// No web page redirects for programmatic requests.
+			for _, accept := range c.Request().Header["Accept"] {
```

Comment:
I'm not very familiar with the HTTP request header libraries. Header is a map right? Do we need to access the "Accept" key using `accept, ok := c.Request().Header["Accept"]`? Or is the Header map always guaranteed to exist and have an "Accept" key?

Reply:
it's good to double check these always. Here this header key could be missing but the type is `type Header map[string][]string` and any nonexistent key would still result in the zero value which would be an empty slice.